### DISCUSSION TITLE
Container runs out of memory (uploading 33Mtriples)

### DISCUSSION THREAD

**Discussion by petterbma** [oxigraph/oxigraph#288] [Q&A] [answered]
Hi,

I am using oxigraph to run as a service with the dockerfile in this repositoy. I am facing an issue when I try to bulk data into the store via:  curl -f -X POST applicaiton/n-triples --upload-file URL/store?default.

I am trying to upload over 33M triples from a single .nt file, but the container runs out of memory. 

Is there a setting I must change, s.t. the oxigraph server uses a persistent disk storage? :) 

Thanks in advance! Really like this graph database!

**Comment by hobofan** [comment, marked as answer]
Looking at the source code, as long as you are providing a `--location some_directory` flag, the oxigraph server will use a persistent disk.

So if you are starting the container according to the command in the README (`docker run --rm -v $PWD/data:/data -p 7878:7878 oxigraph/oxigraph --location /data serve --bind 0.0.0.0:7878`), the container will use persistent disk storage and persist it across containers in a `data` directory inside your current directory.

**Reply by PetterBMarkussen** [reply]
Ok, thanks for quick reply :)

**Reply by PetterBMarkussen** [reply]
Hi again! 

I tried the persistent disk option @hobofan. Now the ./data directory is growing steadly, creating .sst files. Unfortunatley though, it uses more than 28GB of RAM after I send a POST method I mentioned above. 

I am not sure why it uses this much memory. Is the high memory usage related to the parsing of the 8GB .nt file via POST (33 M triples)? If you have any advice how I can index this file via POST, without going above 24 GB RAM, that would be great.

I have tried to search inside the storage/backend directory inside the oxigraph repo, to see if I can tune some parameters relating to the parser, but with no luck.

**Reply by Tpt** [reply]
Hi! This high memory usage is caused by Oxigraph RocksDB storage because the HTTP server is transactional by default and RocksDB keeps all transaction-related data in memory.

A way to load large files to Oxigraph is to use the command line: `docker run --rm -v $PWD/data:/data -p 7878:7878 oxigraph/oxigraph --location /data load --file my_file.nt`. This will do a bulk load of the file into Oxigraph.

**Reply by PetterBMarkussen** [reply]
Thanks for fast answer! 

Ok, so there is no possibility to turn off this transaction-related data @Tpt ? I am using this service as part of my kubernetes cluster, where I use the curl POST to talk between this service and another one. Therefore, I need to upload new tripples store, without shutting down the service.

**Reply by Tpt** [reply]
Not yet sadly. But we should definitly allow it. I have opened #290 about it.
